ere-guests/libs/src/senders.rs

```diff
@@ -1,11 +1,13 @@
 //! This module provies sender recovery helpers for witness generation and block stateless validation.
 
-use alloy_consensus::BlockHeader;
-use alloy_primitives::Address;
-use k256::ecdsa::{VerifyingKey, signature::hazmat::PrehashVerifier};
-use reth_chainspec::EthereumHardforks;
-use reth_ethereum_primitives::{Block, TransactionSigned};
-use reth_primitives_traits::{Block as _, RecoveredBlock};
+use reth_ethereum_primitives::TransactionSigned;
+use serde::{Deserialize, Serialize};
+use serde_with::{Bytes, serde_as};
+
+/// Uncompressed public key wrapper that supports bincode serialization
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct UncompressedPublicKey(#[serde_as(as = "Bytes")] pub [u8; 65]);
```

Comment:
The [UncompressedPublicKey from the stateless crate](https://github.com/paradigmxyz/reth/blob/5887a1596690efff06f7d7b8a812aec120a4a06d/crates/stateless/src/recover_block.rs#L12-L13) doesn't satisfy `Serialize`/`Deserialize`.

This is because length `[u8;65]` for length 65 doesn't have a generic implementation (I think goes up to 32). So, (unfortuantely) I had to create this wrapper. 

Maybe we can think of solving this on the Reth side? Or maybe not a big deal, but anyone trying to use it for guest reading inputs will have to solve this.